bench_method.go

```diff
@@ -36,21 +35,22 @@ type peerTransport struct {
 	peerID int
 }
 
-type benchmarkMethod struct {
-	serializer encoding.Serializer
-	req        *transport.Request
+// benchmarker exposes method to dispatch requests for benchmark.
+type benchmarker interface {
+	// Call dispatches a request using the provided transport.
+	Call(transport.Transport) (time.Duration, error)
```

Comment:
do you want to call method `benchmark` or `benchmarkUnary` ?

Reply:
Renamed it to `benchmarkCaller`